### PR TITLE
Features/30334 heading unit test

### PR DESCRIPTION
**Pull request by DimpleKundanani**
Unit Test Cases for Heading component is ready for code review.

### CODE REVIEW

File: src/components/Heading/tests/Heading.test.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import {mount} from 'enzyme';
```
Comment (kimhjona) [inline src/components/Heading/tests/Heading.test.tsx]: I would put spaces between the brackets  - { mount }

File: src/components/Label/tests/Label.test.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import {mount} from 'enzyme';
```
Comment (kimhjona) [inline src/components/Label/tests/Label.test.tsx]: same as previous comment. I would put spaces between the brackets - { mount }

File: src/components/Label/tests/Label.test.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import Label from '../Label';
+
+describe('Label Suit', () => {
+
+    it('Verify Label is not Hidden', () => {
```
Comment (kimhjona) [inline src/components/Label/tests/Label.test.tsx]: I would make it a complete sentence for readability.

it('should verify Label is not hidden', () => {

File: src/components/Label/tests/Label.test.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    });
+
+    it('Verify Label is  Hidden', () => {
+          const labelWrapper = mount(<Label id="lblid" hidden={false} />);
+          expect(labelWrapper.prop('hidden')).toBe(false);
+    });
+
+    it('Verify type of hidden', () => {
+      const labelWrapper = mount(<Label id="lblid" hidden={false}/>).prop('hidden');
+      expect(typeof (labelWrapper)).toBe('boolean');
```
Comment (kimhjona) [inline src/components/Label/tests/Label.test.tsx]: extra parenthesis. A space between typeof and the value is fine. 
expect(typeof labelWrapper).toBe('boolean');

File: src/components/Label/tests/Label.test.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    it('Verify id', () => {
+      const labelWrapper = mount(<Label id="lblid"/>);
+      expect(labelWrapper.prop('id')).toBe('lblid');
+
+    });
+
+    it('Sets a random id  when none is passed', () => {
+      const labelWrapper = mount(<Label id="lblid"/>).prop('id');
+      expect(typeof (labelWrapper)).toBe('string');
```
Comment (kimhjona) [inline src/components/Label/tests/Label.test.tsx]: same as previous comment

expect(typeof labelWrapper).toBe('string');

File: src/components/Label/tests/Label.test.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+          const labelWrapper = mount(<Label id="lblid" hidden={false} />);
+          expect(labelWrapper.prop('hidden')).toBe(false);
+    });
+
+    it('Verify type of hidden', () => {
+      const labelWrapper = mount(<Label id="lblid" hidden={false}/>).prop('hidden');
+      expect(typeof (labelWrapper)).toBe('boolean');
+    });
+
+    it('Verify id', () => {
```
Comment (kimhjona) [inline src/components/Label/tests/Label.test.tsx]: it('should verify id', () => {

File: src/components/Label/tests/Label.test.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    });
+
+    it('Sets a random id  when none is passed', () => {
+      const labelWrapper = mount(<Label id="lblid"/>).prop('id');
+      expect(typeof (labelWrapper)).toBe('string');
+      expect(labelWrapper).toBeTruthy();
+    });
+
+   it('Children is not defined', () => {
```
Comment (kimhjona) [inline src/components/Label/tests/Label.test.tsx]: same with line 39

File: src/components/Stack/tests/Stack.test.tsx [hunk 8]
```diff
@@ -0,0 +1,254 @@
+import * as React from 'react';
+import {mount} from 'enzyme';
```
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: { mount }

File: src/components/Stack/tests/Stack.test.tsx [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+import Stack from '../Stack';
+import Badge from '../';
```
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: not sure where this is being imported from. did you mean - import Badge from './../../Badge';   ?

File: src/components/Stack/tests/Stack.test.tsx [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+import Item from '../Item';
+import Heading from '../';
```
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: same as previous comment. import Heading from '../../Heading'

File: src/components/Stack/tests/Stack.test.tsx [hunk 11]
```diff
@@ -0,0 +1,254 @@
...
+describe('Stack Component Suit', () => {
+    describe('Test children property', () => {
+        it('should verify children property is defined', () => {
+            const stackWrapper = mount(
+                                        <Stack>
+                                          <Badge>Paid</Badge>
+                                          <Badge>Fulfilled</Badge>
+                                        </Stack>,
+                                 );
+            expect(stackWrapper.childAt(0).prop('children').props.children).toBe('Paid');
```
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: just a tiny thing. I would probably push line 12 up to line 11 after the parenthesis. Something like 

+            const stackWrapper = mount(<Stack>

Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: same for line 30-31

File: src/components/Stack/tests/Stack.test.tsx [hunk 12]
```diff
@@ -0,0 +1,254 @@
...
+        });
+
+         it('should verify type of vertical property ', () => {
+            const stackWrapper = mount(
+                                        <Stack vertical>
+                                          <Badge>Paid</Badge>
+                                          <Badge>Fulfilled</Badge>
+                                        </Stack>,
+                                 ).prop('vertical');
+            expect(typeof (stackWrapper)).toBe('boolean');
```
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: don't need the parenthesis
expect(typeof stackWrapper).toBe('boolean');

File: src/components/Stack/tests/Stack.test.tsx [hunk 13]
```diff
@@ -0,0 +1,254 @@
...
+import Heading from '../';
+
+describe('Stack Component Suit', () => {
+    describe('Test children property', () => {
+        it('should verify children property is defined', () => {
+            const stackWrapper = mount(
+                                        <Stack>
+                                          <Badge>Paid</Badge>
+                                          <Badge>Fulfilled</Badge>
+                                        </Stack>,
```
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: not sure why the comma is there. mount doesn't take a second argument (this applies to every single one)

File: src/components/Stack/tests/Stack.test.tsx [hunk 14]
```diff
@@ -0,0 +1,254 @@
...
+            const stackWrapper = mount(<Stack/>);
+            expect(stackWrapper.prop('children')).toBeUndefined();
+        });
+    });
+
+    describe('Test vertical property', () => {
+        it('should verify vertical property is set', () => {
+            const stackWrapper = mount(
+                                        <Stack vertical>
+                                          <Badge>Paid</Badge>
```
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: why do you include the badges for every single test if you never test anything about them? not that it changes anything, just seems redundant.

File: src/components/Stack/tests/Stack.test.tsx [hunk 15]
```diff
@@ -0,0 +1,254 @@
...
+                                        </Stack>,
+                                 );
+            expect(stackWrapper.prop('distribution')).toBeUndefined();
+        });
+    });
+
+    describe('Test stack with item', () => {
+        it('should verify stack where a single item fills the remaining space', () => {
+            const stackWrapper = mount(
+                                        <Stack >
```
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: Take out the extra spaces

<Stack>

Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: <Stack>

File: src/components/Tooltip/tests/Tooltip.test.tsx [hunk 16]
```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import {mount} from 'enzyme';
```
Comment (kimhjona) [inline src/components/Tooltip/tests/Tooltip.test.tsx]: { mount }

File: src/components/Heading/tests/Heading.test.tsx (hunk 1 above)
Comment (kimhjona) [inline src/components/Heading/tests/Heading.test.tsx]: { mount }

File: src/components/Label/tests/Label.test.tsx (hunk 2 above)
Comment (kimhjona) [inline src/components/Label/tests/Label.test.tsx]: { mount }

File: src/components/Label/tests/Label.test.tsx (hunk 5 above)
Comment (kimhjona) [inline src/components/Label/tests/Label.test.tsx]: don't need parenthesis - typeof labelWrapper

File: src/components/Stack/tests/Stack.test.tsx (hunk 9 above)
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: haven't seen this before. did you mean import Badge from '../../Badge';  ?

Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: same for line 6

File: src/components/Stack/tests/Stack.test.tsx [hunk 17]
```diff
@@ -0,0 +1,254 @@
...
+import {mount} from 'enzyme';
+import Stack from '../Stack';
+import Badge from '../';
+import Item from '../Item';
+import Heading from '../';
+
+describe('Stack Component Suit', () => {
+    describe('Test children property', () => {
+        it('should verify children property is defined', () => {
+            const stackWrapper = mount(
```
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: I would push line 12 to line 11. const stackWrapper = mount(<Stack>

Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: this applies for all of them

File: src/components/Stack/tests/Stack.test.tsx (hunk 13 above)
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: Not sure what the comma is doing here

Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: applies for all of them

File: src/components/Stack/tests/Stack.test.tsx (hunk 14 above)
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: Why do you keep using the Badge component, but never actually test it? All your tests would work without line 32 and 33. Doesn't break anything, but seems redundant.

File: src/components/Stack/tests/Stack.test.tsx (hunk 15 above)
Comment (kimhjona) [inline src/components/Stack/tests/Stack.test.tsx]: you don't need the space between the k and the >. This applies to all the components with the extra space.

File: src/components/Tooltip/tests/Tooltip.test.tsx [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+import Tooltip from '../Tooltip';
+import Link from '../';
```
Comment (kimhjona) [inline src/components/Tooltip/tests/Tooltip.test.tsx]: did you mean import Link from '../../../Link'; ?

File: src/components/Tooltip/tests/Tooltip.test.tsx [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+
+
+describe('Tooltip Component Suit', () => {
+    describe('Test children property', () => {
+        it('should verify children property is defined as link', () => {
+            const tooltipWrapper = mount(
```
Comment (kimhjona) [inline src/components/Tooltip/tests/Tooltip.test.tsx]: same as previous comment. I would push line 11 up to line 10 and take out the comma at the end of line 13

File: src/components/Tooltip/tests/Tooltip.test.tsx [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+    describe('Test content property', () => {
+        it('should verify content property is available', () => {
+            const tooltipWrapper = mount(<Tooltip content="This order has shipping labels1."/>);
+            expect(tooltipWrapper.prop('content')).toBe('This order has shipping labels1.');
+        });
+    });
+
+    describe('Test active property', () => {
+        it('should verify active property is available', () => {
+            const tooltipWrapper = mount(<Tooltip content="This order has shipping labels1." active/>);
```
Comment (kimhjona) [inline src/components/Tooltip/tests/Tooltip.test.tsx]: you need a space when you use self-closing brackets. 

< />
<>

**Comment by kimhjona** [comment]
This is really complete and good quality. I have a few suggestions for mostly readability and some functionality

File: src/components/Heading/tests/Heading.test.tsx [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+        });
+    });
+
+    describe('Test element property', () => {
+        it('should verify element property when set as h1', () => {
+            const headingWrapper = mount(<Heading element="h1">Online store dashboard</Heading>);
+            expect(headingWrapper.prop('element')).toBe('h1');
+        });
+
+         it('should verify element property when set as h2', () => {
```
Comment (kimhjona) [inline src/components/Heading/tests/Heading.test.tsx]: lines 24 to 52 are redundant

Comment (DimpleKundanani) [inline src/components/Heading/tests/Heading.test.tsx]: I have updated test cases and added unit test cases for how elements are rendered.So not removing that unit test cases.
Please review that.

**Pull request merged**